lib/src/decorator.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import 'dart:html';
+
+import 'package:meta/meta.dart';
+import 'package:platform_detect/platform_detect.dart';
+import 'package:platform_detect/src/support.dart';
+
+/// The CSS class that will be used to indicate that [decorateRootNodeWithPlatformClasses]
+/// has already been invoked for a given root node.
+const String _decorationCompleteClassName = 'pd--decorated';
+
+/// The CSS class that will be used to represent the current [browser].
+final String _browserClassName =
+    browser.className ?? nameToClassName(browser.name);
+
+/// The CSS class that will be used to represent the current [operatingSystem].
+final String _osClassName = nameToClassName(operatingSystem.name);
+
+/// The string that will be prepended to the CSS class generated by [getPlatformClasses]
+/// for [_browserClassName].
+const String _browserClassPrefix = 'ua-';
+
+/// The string that will be prepended to the CSS class generated by [getPlatformClasses]
+/// for [_osClassName].
+const String _osClassPrefix = 'os-';
+
+/// The string that will be prepended to the CSS class generated by [getPlatformClasses]
+/// for [_osClassName].
+const String _versionRangeClassPrefix = 'lt-';
+
+/// The string that will be prepended to the CSS class generated by [getPlatformClasses]
+/// for [Feature.name] if [Feature.isSupported] is `false`.
+const String _featureSupportNegationClassPrefix = 'no-';
+
+/// The first major release after the current one.
+@visibleForTesting
+final int minVersionRange = browser.version.major + 1;
+
+/// The number of major releases above [minVersionRange]
+/// that [_getBrowserVersionClasses] should produce CSS classes for.
+@visibleForTesting
+const int maxVersionRange = 2;
```

Comment:
These variable names are a bit confusing since `minVersionRange`/`maxVersionRange` have different semantic meanings (one's a bound, one's an offset).

What about something like `nextVersion` and `decoratedNextVersionCount`?